### ISSUE TITLE
AWS S3 data source URI versus Glue Data Catalog

### ISSUE THREAD

**Issue by wajda** [AbsaOSS/spline-spark-agent#157]
For the reference:
https://issues.apache.org/jira/browse/ATLAS-2760

**Comment by ASchwad** [comment]
We use Spline with Glue and S3 the following way:

The [Glue Data Catalog](https://docs.aws.amazon.com/glue/latest/dg/populate-data-catalog.html) uses databases and tables as abstractions, where one database consists of n tables.
In S3, this relationship is displayed within the S3 bucket uri.
 
e.g. `S3://my-database/my-table/p_ingest_data=2020-09-25`
where `p_ingest_data=2020-09-25` is the partition.

But these uri elements do not necessarily match the Glue Data Catalog Database and table name, which is a important requirement for us to match the catalog entities to their execution plans. Luckily, the identifiers are contained out-of-the-box within the operation params in the execution plan. 

Here is an example of a write operation into a S3 Bucket via a Glue Job.

```json
      "write": {
          "outputSource": "s3://my-database-xyz/my-table-abc",
          "append": false,
          "id": 0,
          "childIds": [
              1
          ],
          "params": {
              "table": {
                  "identifier": {
                      "table": "my_table",
                      "database": "my_database"
                  },
                  "storage": {
                      "locationUri": "s3://my-database-xyz/my-table-abc",
                      "compressed": false
                  }
              }
          },
          "extra": {
              "name": "CreateDataSourceTableAsSelectCommand",
              "destinationType": "parquet"
          }
      },
```

Maybe these insights help you for the implementation.

**Comment by wajda** [comment]
Thanks. Yes, that's a more generic data source identity problem that will be partially addressed in AbsaOSS/spline#689

**Comment by wajda** [comment]
Closing in favor of https://github.com/AbsaOSS/spline/issues/689
